### ISSUE TITLE
The examples don't seem to be assemble-able

### ISSUE THREAD

**Issue by tekknolagi** [KCreate/stackvm#5]
For example:

```
willow% ./bin/stackvm build examples/loops.asm -s -o loops.bc    
Error: sub expected 2 arguments, got 3 at /home/max/Documents/Dev/stackvm/examples/loops.asm:40:3
```

**Comment by KCreate** [comment]
Yes, that's a recent change I made and I forgot to update the examples. You can rely on the design documents, as I always update them first before changing the code.

I'll fix that ASAP.

**Comment by tekknolagi** [comment]
I made what I thought were appropriate fixes to the code (remove the dst register, etc) but now I get an "Index out of bounds" error at runtime. Is this worth a separate issue or is it related to the examples being behind?

**Comment by KCreate** [comment]
No `Index out of bounds` shouldn't be a normal error. Could you provide a stack trace?

Change [these lines](https://github.com/KCreate/stackvm/blob/master/src/command.cr#L124) to the following:

```crystal
if err
  puts err
  puts err.backtrace
  return
end
```

That'll give you a backtrace for the error.

**Comment by KCreate** [comment]
To answer your original question, please open another issue for this.

**Comment by KCreate** [comment]
All examples were fixed in 70a02862b1e9df675dbb7535222daee70d1d483a.
The `Index out of bounds` was fixed in d043a85057604f600abb508821a58451da958160
